internal/configsource/vaultconfigsource/README.md

```diff
@@ -0,0 +1,88 @@
+# Vault Config Source
+
+Use the [Vault](https://www.vaultproject.io/) config source to retrieve data from
+Vault and inject it into your collector configuration. It supports:
+
+- [Dynamic secrets](https://www.vaultproject.io/);
+- [Key/Value V1 lease hints](https://www.vaultproject.io/docs/secrets/kv/kv-v1);
+- [Key/Value V2 metadata polling](https://www.vaultproject.io/docs/secrets/kv/kv-v2);
+
+## Configuration
+
+Under the `config_sources:` use `vault:` or `vault/<name>:` to create a Vault config
+source Under the latter section, the following parameters are available to customize
+Vault config sources:
+
+```yaml
+config_sources:
+  vault:
+    # endpoint is the Vault server address. It is equivalent to the Vault tool
+    # environment variable VAULT_ADDR.
+    endpoint: http://localhost:8200
+    # path is the Vault path to the secret location.
+    path: secret/data/kv
+    # poll_interval is used only for non-dynamic V2 K/V secret stores. It is
+    # the interval in which the config source will check for changes on the
+    # data on the given Vault path. Defaults to 1 minute if not specified.
+    poll_interval: 90s
+    # auth is a section used to indicate the authentication method to be used.
+    # Exactly one method must be specified, it must be one of the following:
+    # "token", "iam", or "gcp".
+    auth:
```

Comment:
There is an initiative at Otel Collector to define extensible auth mechanism for outgoing connections. It would be good to see if we can reuse it here. It is not yet ready at Otel so hard to tell but once it is I think we should evaluate it. This means that the config here may need to break in the near future, but I think that's OK.